torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -487,9 +504,12 @@ void ProcessGroupAgent::listenLoop() {
 }
 
 void ProcessGroupAgent::pollTimedOutRPCs() {
-  while (!shutdown_.load()) {
-    std::chrono::milliseconds sleepTime;
+  while (true) {
     std::unique_lock<std::mutex> lock{futureMutex_};
```

Comment:
Just make sure I understand this correctly. If this is before the setting `rpcRunning_` in `shutdown`, then `futureTimeoutCV_` will be notified properly. If this is after, then this would return as `rpcRunning_.load() == false`, right?

Reply:
Yes - for the first case you mentioned, this thread would drop the lock when it waits and would be appropriately notified by shutdown. For the 2nd case, we would break as `rpcRunning = false`.